### PR TITLE
Ssh issues

### PR DESCRIPTION
**Pull request by hossamabdullah**
#### What this PR resolves:
create a machine is stable using ssh connections

**Version**: 9.3.0

### CODE REVIEW

File: JumpScale9/clients/ssh/SSHClient.py [hunk 1]
```diff
@@ -169,7 +179,8 @@ def _connect(self):
                     banner_timeout=3.0)
                 self.logger.info("connection ok")
                 return self._client
-
+            except BadAuthenticationType as e:
+                raise e
```
Comment (ahussein) [inline JumpScale9/clients/ssh/SSHClient.py]: This is not needed, it does not make sense to just catch the exception and re-raise it.

File: JumpScale9/clients/ssh/SSHClient.py [hunk 2]
```diff
@@ -340,7 +351,7 @@ def rsync_down(self, source, dest, source_prefix="", recursive=True):
     def prefab(self):
         if not self.usesproxy and self._prefab is None:
             executor = j.tools.executor.getSSHBased(
-                self.addr, self.port, self.login, self.passwd)
+                addr=self.addr, port=self.port, timeout=self.timeout)
```
Comment (ahussein) [inline JumpScale9/clients/ssh/SSHClient.py]: Why this change is needed ?

File: JumpScale9/clients/ssh/SSHClient.py [hunk 3]
```diff
@@ -370,15 +381,17 @@ def portforwardKill(self, localport):
 
     def SSHAuthorizeKey(
             self,
-            keyname):
+            sshkey_name,
+            sshkey_path):
         """
         this required ssh-agent to be loaded !!!
-        the keyname is the name of the key as loaded in ssh-agent
+        the sshkey_name is the name of the key as loaded in ssh-agent
```
Comment (ahussein) [inline JumpScale9/clients/ssh/SSHClient.py]: Please update the doc sting, right now it just does not make sense.

File: JumpScale9/clients/ssh/SSHClient.py (hunk 1 above)
Comment (hossamabdullah) [reply JumpScale9/clients/ssh/SSHClient.py]: it will be caught by another generalized exception and will be raised as a RuntimeError.
I need to keep the type of the exception.

File: JumpScale9/clients/ssh/SSHClient.py (hunk 2 above)
Comment (hossamabdullah) [reply JumpScale9/clients/ssh/SSHClient.py]: as the function parameters have been changed
it will pass self.login for timeout parameter
